src/pg/sql/42_observatory_exploration.sql

```diff
@@ -181,6 +181,90 @@ BEGIN
 END
 $$ LANGUAGE plpgsql;
 
+CREATE OR REPLACE FUNCTION cdb_observatory._OBS_GetNumerators(
+  bounds GEOMETRY DEFAULT NULL,
+  section_tags TEXT[] DEFAULT NULL,
+  subsection_tags TEXT[] DEFAULT NULL,
+  other_tags TEXT[] DEFAULT NULL,
+  ids TEXT[] DEFAULT NULL,
+  name TEXT DEFAULT NULL,
+  denom_id TEXT DEFAULT NULL,
+  geom_id TEXT DEFAULT NULL,
+  timespan TEXT DEFAULT NULL
+) RETURNS TABLE (
+  numer_id TEXT,
+  numer_name TEXT,
+  numer_description TEXT,
+  numer_weight NUMERIC,
+  numer_license TEXT,
+  numer_source TEXT,
+  numer_type TEXT,
+  numer_aggregate TEXT,
+  numer_extra JSONB,
+  numer_tags JSONB,
+  valid_denom BOOLEAN,
+  valid_geom BOOLEAN,
+  valid_timespan BOOLEAN
+) AS $$
+DECLARE
+  where_clause_elements TEXT[];
+  where_clause TEXT;
+BEGIN
+  section_tags := COALESCE(section_tags, (ARRAY[])::TEXT[]);
+  subsection_tags := COALESCE(subsection_tags, (ARRAY[])::TEXT[]);
+  other_tags := COALESCE(other_tags, (ARRAY[])::TEXT[]);
+  ids := COALESCE(ids, (ARRAY[])::TEXT[]);
+  name := COALESCE(name, '');
+  denom_id := COALESCE(denom_id, '');
+  geom_id := COALESCE(geom_id, '');
+  timespan := COALESCE(timespan, '');
+  where_clause_elements := (ARRAY[])::TEXT[];
+  where_clause := '';
+
+  IF cardinality(section_tags) > 0 THEN
+    where_clause_elements := array_append(where_clause_elements, format($data$numer_tags ?| '%s'$data$, section_tags));
+  END IF;
+  IF cardinality(subsection_tags) > 0 THEN
+    where_clause_elements := array_append(where_clause_elements, format($data$numer_tags ?| '%s'$data$, subsection_tags));
+  END IF;
+  IF cardinality(other_tags) > 0 THEN
+    where_clause_elements := array_append(where_clause_elements, format($data$numer_tags ?| '%s'$data$, other_tags));
+  END IF;
+  IF cardinality(ids) > 0 THEN
+    where_clause_elements := array_append(where_clause_elements, format($data$numer_id IN (array_to_string('%s'::text[], ','))$data$, ids));
+  END IF;
+  IF name IS NOT NULL AND name != '' THEN
+    where_clause_elements := array_append(where_clause_elements, format($data$numer_name ilike '%%%s%%'$data$, name));
```

Comment:
I did it with full text search but we need something that let us use partial words. I've tried with trigrams too but i doesn't work well for 3 letters and we need to install a new extensión `pg_trgm` so I decided to use the LIKE operator because we're dealing with just 17k rows at most.